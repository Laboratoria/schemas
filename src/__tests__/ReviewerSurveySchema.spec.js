const mongoose = require('mongoose/browser');
const { ReviewerSurveySchema, ReviewQuestionSchema } = require('../../')(mongoose);

describe('ReviewerSurveySchema', () => {
  it('should fail validation when fields missing', () => {
    const question = new mongoose.Document({}, ReviewQuestionSchema);
    const doc = new mongoose.Document({
      question: question._id,
    }, ReviewerSurveySchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should pass sync validation ...', () => {
    const question = new mongoose.Document({}, ReviewQuestionSchema);
    const doc = new mongoose.Document({
      question: question._id,
      version: '2.0.0',
    }, ReviewerSurveySchema);
    expect(doc.validateSync()).toBe(undefined);
  });

  it('should validate good question', () => {
    const question = new mongoose.Document({}, ReviewQuestionSchema);
    const doc = new mongoose.Document({
      question: question._id,
      version: '2.0.0',
    }, ReviewerSurveySchema);

    return doc.validate();
  });
});
