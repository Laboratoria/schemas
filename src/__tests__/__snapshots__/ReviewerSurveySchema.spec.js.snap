// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewerSurveySchema should fail validation when fields missing 1`] = `
Object {
  "questions": [ValidatorError: Path \`questions\` is required.],
}
`;

exports[`ReviewerSurveySchema should fail validation when multiple-choice question is missing options 1`] = `"Validation failed: questions: Options is required when type is multiple-choice"`;

exports[`ReviewerSurveySchema should fail validation when question is missing slug 1`] = `"Validation failed: questions.0.slug: Path \`slug\` is required."`;

exports[`ReviewerSurveySchema should fail validation when question options not a number 1`] = `
Object {
  "questions": [Error: Options is required when type is multiple-choice],
  "questions.0.options": [CastError: Cast to Number failed for value "hola" at path "options"],
}
`;

exports[`ReviewerSurveySchema should fail validation when wrong type 1`] = `
Object {
  "questions.0.type": [ValidatorError: \`other\` is not a valid enum value for path \`type\`.],
}
`;
